src/components/modals/MigrateAccounts/index.js

```diff
@@ -0,0 +1,201 @@
+import type { StepProps as DefaultStepProps, Step } from 'components/base/Stepper'
+
+import StepOverview, { StepOverviewFooter } from './steps/01-step-overview'
+import StepConnectDevice, { StepConnectDeviceFooter } from './steps/02-step-connect-device'
+import StepCurrency, { StepCurrencyFooter } from './steps/03-step-currency'
+import { getCurrentDevice } from '../../../reducers/devices'
+import { replaceAccounts } from '../../../actions/accounts'
+import { closeModal } from '../../../reducers/modals'
+
+const createSteps = () => {
+  const onBack = ({ transitionTo }: StepProps) => {
+    transitionTo('overview')
+  }
+  return [
+    {
+      id: 'overview',
+      component: StepOverview,
+      footer: StepOverviewFooter,
+    },
+
+    {
+      id: 'device',
+      component: StepConnectDevice,
+      footer: StepConnectDeviceFooter,
+      onBack,
+    },
+    {
+      id: 'currency',
+      component: StepCurrency,
+      footer: StepCurrencyFooter,
+    },
+  ]
+}
+
+type StepId = 'overview'
+type ScanStatus = 'idle' | 'scanning' | 'error' | 'finished'
+export type StepProps = DefaultStepProps & {
+  migratableAccounts: { [key: string]: Account[] },
+  migratedAccounts: { [key: string]: Account[] },
+  accounts: Account[],
+  totalMigratableAccounts: number,
+  currency: ?CryptoCurrency,
+  setScanStatus: (ScanStatus, ?Error) => string,
+  addMigratedAccount: (CryptoCurrency, Account) => void,
+  device: ?Device,
+  nextCurrency: () => void,
+  flushMigratedAccounts: () => void,
+  setAppOpened: (isAppOpened: boolean) => void,
+  scanStatus: ScanStatus | string,
+  isAppOpened: boolean,
+  onCloseModal: () => void,
+}
+
+type State = {
+  stepId: StepId | string,
+  isAppOpened: boolean,
+  currency: ?CryptoCurrency,
+  scanStatus: ScanStatus | string,
+  migratedAccounts: { [key: string]: Account[] },
+  err: ?Error,
+}
+
+const INITIAL_STATE = {
+  stepId: 'overview',
+  currency: null,
+  isAppOpened: false,
+  migratedAccounts: {},
+  scanStatus: '',
+  err: null,
+}
+
+class MigrateAccounts extends PureComponent<*, State> {
+  state = INITIAL_STATE
+  STEPS = createSteps()
+
+  handleStepChange = (step: Step) => this.setState({ stepId: step.id })
+  handleSetAppOpened = (isAppOpened: boolean) => this.setState({ isAppOpened })
+  handleAddMigratedAccounts = (currency, account) => {
+    this.setState(prevState => ({
+      migratedAccounts: {
+        ...prevState.migratedAccounts,
+        [currency.id]: [...(prevState.migratedAccounts[currency.id] || []), account],
+      },
+    }))
+  }
+  handleSetScanStatus = (scanStatus: string, err: ?Error = null) => {
+    if (err) {
+      logger.critical(err)
+    }
+    this.setState({ scanStatus, err })
+  }
+
+  handleNextCurrency = () => {
+    const { migratableAccounts } = this.props
+    const { migratedAccounts } = this.state
+    const currencyId = Object.keys(migratableAccounts).find(
+      d => !Object.keys(migratedAccounts).includes(d),
```

Comment:
i'm a bit lost in how this select the "next currency". My current intuition is it will never reach a "third" currency if some of the account was not migrated? maybe i'm missing something.
at no point you have an iteration on the currency so i'm a bit 🤔  how it works

should it work by looking at the indexOf in the migratable per currency and just take the next index?

I think there is something simpler to use in term of codebase.
I wonder if this can already work:

```js
const ids = Object.keys(migrableAccounts)
const nextCurrencyId = ids[ ids.indexOf(currentCurrencyId)+1 ]
const currency = nextCurrencyId ? getCryptoCurrencyById(nextCurrencyId) : null
```

interesting fact: `currency===null` would mean you are done on the "currency iteration", then you can check if migratableAccounts is empty to know if others remain.